### PR TITLE
Add Arraylist support in WriteableArray

### PR DESCRIPTION
**Pull request by cacheung**
Add Arraylist support in WriteableArray for Android.
If response payload has arraylist in it, it can be extracted now. 

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [ ] I have signed the [Adobe Open Source CLA](http://opensource.adobe.com/cla.html).
- [ ] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [ ] I have read the **CONTRIBUTING** document.
- [ ] I have added tests to cover my changes.
- [ ] All new and existing tests passed.

### CODE REVIEW

File: packages/edge/android/src/main/java/com/adobe/marketing/mobile/reactnative/edge/RCTAEPEdgeArrayUtil.java [hunk 1]
```diff
@@ -81,4 +82,31 @@ static WritableArray toWritableArray(Object[] array) {
...
+            } else if (value instanceof Boolean) {
+                writableArr.pushBoolean((Boolean) value);
+            } else if (value instanceof Double) {
+                writableArr.pushDouble((Double) value);
+            } else if (value instanceof Integer) {
+                writableArr.pushInt((Integer) value);
+            } else if (value instanceof String) {
+                writableArr.pushString((String) value);
+            } else if (value instanceof Map) {
+                writableArr.pushMap(RCTAEPEdgeMapUtil.toWritableMap((Map<String, Object>) value));
```
Comment (emdobrin) [inline packages/edge/android/src/main/java/com/adobe/marketing/mobile/reactnative/edge/RCTAEPEdgeArrayUtil.java:104]: we should add the check for instanceof ArrayList to L105 and L78 too

**Comment by yangyansong-adbe** [comment]
> @yangyansong-adbe @emdobrin Do we need to add ArrayList check for other extensions if they are using ArrayUtil?

@cacheung Yes, I think we need to update this utility method in other extensions.

File: packages/edge/android/src/main/java/com/adobe/marketing/mobile/reactnative/edge/RCTAEPEdgeArrayUtil.java (hunk 1 above)
Comment (cacheung) [reply packages/edge/android/src/main/java/com/adobe/marketing/mobile/reactnative/edge/RCTAEPEdgeArrayUtil.java:106]: Tested with List and it also works.  Seems like Changing to List is better?

**Review by emdobrin — APPROVED** [review]

**Review by yangyansong-adbe — APPROVED** [review]

**Comment by cacheung** [comment]
> @cacheung can you check why CI is not triggered for this PR?

CI is working now after @yangyansong-adbe enabled the setting for running CI for each PR.

**Pull request merged**